test/apps/iodemo/ucx_wrapper.h

```diff
@@ -86,15 +86,13 @@ class UcxContext {
 
 protected:
 
-    // Called when new connection is created on server side
-    virtual void dispatch_new_connection(UcxConnection *conn);
-
     // Called when new IO message is received
     virtual void dispatch_io_message(UcxConnection* conn, const void *buffer,
-                                     size_t length);
+                                     size_t length) = 0;
 
     // Called when there is a fatal failure on the connection
-    virtual void dispatch_connection_error(UcxConnection* conn);
+    virtual void dispatch_connection_error(UcxConnection* conn,
+                                           const char *reason) = 0;
```

Comment:
do not modify ucx_wrapper
have 1 more function in DemoClient only which is called both from dispatch_connecton_error, and laso from timeout flow, which gets the reason as parameter

Reply:
I'm not sure that completely understand you, `dispatch_connection_error` is also called from `UcxContext::progress_failed_connections` so we have to know reason here too.

Reply:
```
class IOdemoCommon: public UcxContext
protected:
/// @override
virtual void dispatch_connection_error(UcxConnection *conn) {
  terminate_connection(conn, ucs_status_string(conn->get_status()));
}

void terminate_connection(conn, char* reason) {
 LOG << ... << reason;
 .. all cleanup code
}
}

class P2cclient: public IOdemoCommon {
void close_uncompleted_servers(const char *reason) {
 for (...) {
  if (get_num_uncompleted(i) > 0) {
   terminate_connection(server_info[i].conn, reason)
 }
}
}

wait_for_completions() {
...
}

```